spec/features/admin/applications/show_spec.rb

```diff
@@ -0,0 +1,144 @@
+require "rails_helper"
+
+RSpec.describe "Admin Application Show Page", type: :feature do
+
+  describe "As a visitor" do
+    describe "When I visit '/admin/applications/:id'" do
+      before(:each) do
+        @shelter_1 = Shelter.create!(name: "Aurora shelter", city: "Aurora, CO", foster_program: false, rank: 9)
+        @pet_1 = Pet.create!(adoptable: true, age: 1, breed: "doberman", name: "Auggie", shelter_id: @shelter_1.id)
+        @pet_2 = Pet.create!(adoptable: true, age: 6, breed: "pug", name: "Rue", shelter_id: @shelter_1.id)
+        @pet_3 = Pet.create(adoptable: true, age: 3, breed: "boxer", name: "Ann", shelter_id: @shelter_1.id)
+        @application_1 = Application.create!(name: "Julie Johnson", street_address: "201 Main Street", city: "Seattle", state: "WA", zip_code: "75250", description: "I love dogs!", status: "In Progress")
+        @application_2 = Application.create!(name: "Steve Smith", street_address: "705 Olive Lane", city: "Omaha", state: "NE", zip_code: "98253", description: "Emotional support animal.", status: "In Progress")
+        @applicant_1 = PetApplication.create(application_id: @application_1.id, pet_id: @pet_1.id)
+        @applicant_2 = PetApplication.create(application_id: @application_1.id, pet_id: @pet_3.id)
+        @applicant_3 = PetApplication.create(application_id: @application_2.id, pet_id: @pet_2.id)
+        @applicant_4 = PetApplication.create(application_id: @application_2.id, pet_id: @pet_1.id)
+        @applicant_5 = PetApplication.create(application_id: @application_2.id, pet_id: @pet_3.id)
+      end 
+
+      it "For every pet that the application is for, I see a button to approve the application for that specific pet" do
+
+        visit "admin/applications/#{@application_1.id}"
+
+        within("#Auggie") do
+          expect(page).to have_content(@pet_1.name)
+          expect(page).to have_button("Approve")
+        end 
+
+        within("#Ann") do
+          expect(page).to have_content(@pet_3.name)
+          expect(page).to have_button("Approve")
+        end
+      end
```

Comment:
Nice job using `within` blocks to ensure that what you are wanting is showing up where you expect it to.